### DISCUSSION TITLE
Build bevy game

### DISCUSSION THREAD

**Discussion by wotori** [bevyengine/bevy#4176] [Q&A] [answered]
Could anybody point me how to build game for production under any os?
I am trying under Ubuntu currently with `cargo build release` and when I run `./target/release/bevy-app` everything seems fine but the app lost assets dir that is in the root project`s folder. How to deal with It?
```
WARN bevy_asset::asset_server: encountered an error while reading an asset: path not found: /home/wotori/git/wotori/ekza/ekza-bevy-engine/target/release/assets/models/model.gltf
```
And In the future there should be possibility to build for ios and android right? How soon?

**Comment by bjorn3** [comment, marked as answer]
The `assets` dir needs to be either next to the executable or the `CARGO_MANIFEST_DIR` env var needs to be set to the directory containing the `assets` dir. `cargo run` automatically sets this env var to the directory containing `Cargo.toml` for you, but if you manually execute the game you need to set it yourself.

ios and android are in principle supported I believe, but there are still some kinks around suspending apps. I believe there is a PR for fixing this.

**Reply by wotori** [reply]
Yeah, you right. After moving asset folder to executable`s root folder everything started work as expected. But Is It possible to compile the project with single executable file that doesn't require any additional files?

**Reply by wotori** [reply]
`cargo build release` - Is the only option for building project for production?
So I can compress It and share with end-users? Or Is there a better pipeline?

**Reply by bjorn3** [reply]
There are third party plugins like https://crates.io/crates/bevy-embasset that allow embedding assets into the main executable.

**Reply by mockersf** [reply]
You can check https://bevyengine.org/assets/#asset-loading for options, like embedding your assets, loading them from the web, ...
